Add HTMLLinkElement.href (#31462)
mdn/content: files/en-us/web/api/htmllinkelement/href/index.md

---
title: "HTMLLinkElement: href property"
short-title: href
slug: Web/API/HTMLLinkElement/href
page-type: web-api-instance-property
browser-compat: api.HTMLLinkElement.href
---

{{APIRef("HTML DOM")}}

> The **`HTMLLinkElement.href`** property of the {{domxref("HTMLLinkElement")}} interface contains a string that is the URL associated with the link.

Reviewer: ```suggestion
The **`href`** property of the {{domxref("HTMLLinkElement")}} interface contains a string that is the URL associated with the link.
```

I think the style of *`property`* is better then *`interface.property`*

Author: Oh yes, it was a bad cut and past from my side.

> It reflects the `href` attribute of the {{HTMLElement("link")}} element and is the empty string (`""`) if there are none.

Reviewer 2: ```suggestion
It reflects the `href` attribute of the {{HTMLElement("link")}} element. If the element does not have an `href` attribute, then this property's value is the empty string (`""`).
```

> **Note:** Either this property or {{domxref("HTMLLinkElement.imageSrcset", "imageSrcset")}} must not be empty.

Reviewer 2: ```suggestion
> **Note:** Either this property or {{domxref("HTMLLinkElement.imageSrcset", "imageSrcset")}} must not be empty.
```

This is odd though because isn't this restriction about the HTML element, not the HTMLLinkElement interface? I mean I can do this:

```js
const s = document.createElement("link") 
s.href  // ""
s.imageSrcset  // ""
```
